Draft: Prochaine version de LinuxFr avec Rails8

Trim opened the pull request:
Je crée cette pull request en brouillon pour suivre le plan de [ce commentaire](https://github.com/linuxfrorg/linuxfr.org/pull/378#issuecomment-1982652704).

-----

Hello,

J'ai regardé en vitesse les premiers fichiers et ça à l'aire bien (decaffeinate semble avoir bien marché, merci pour l'ajout des commits avec les fixes des suggestions).

Comme il y a beaucoup de changements tout comme la migration de rails prévue par @echarp, ça va demander beaucoup de temps pour tester.

Je propose de fusionner les 2 développements ensemble et de ne faire les tests qu'une seul fois sur tout le site.

Le plan serait de:

* [x] Créer une branche "rails-7" depuis la master actuelle
* [x] Fusionner la PR #378 dans celle-ci
* [x] Fusionner la PR #375
* [x] Fusionner ma branche qui met à jour les Dockerfile pour Ruby 3 / Rails 7
* [x] Publier une nouvelle version de board-linuxfr avec la mise à jour de Goliath (~~je vais publier sur le repository de Github~~ j'ai publié sur rubygems.org avec le nom `board-sse-linuxfr.org`) pour que ça fonctionne avec Ruby 3
* [x] Ajouter un commit pour utiliser cette nouvelle version de board-linuxfr dans les Docker
* [x] Corriger l'insertion des images pour utiliser le service img-linuxfr (la méthode wikify ne semble plus fonctionner, les images des liens restent sur la source)
* [x] Utiliser par défaut le port `9000` pour ne plus avoir besoin de CAP réseau élevée (#395)
* [x] Mettre à jour le fichier `INSTALL.md`
* [ ] Faire le merge sur master (ou au moins faire un rebase et geler master)
* [ ] Installer tout ça sur le domaine alpha.linuxfr.org
* [ ] Demander aux différents acteurs de tester (admin, animateur, moderateur, mainteneur, visiteur) sur cette instance.

nud committed:
Remove CoffeeScript, use plain Javascript instead

CoffeeScript is (mostly) dead these days and, while it was advertised
with Rails 5 and older, it is not supported anymore by newer version of
Rails. Using plain Javascript might also make it easier for new
contributors to fix issues as nobody is familiar with CoffeeScript these
days.

In this commit we run `decaffeinate` on all the `.coffee` files and
nothing else.

See https://decaffeinate-project.org/

Suivi #3034

echarp committed:
Suppression des inline/dataURI de certains boutons.

Dépendant de la gemme rails-sass-images qui n'est plus développée

echarp committed:
Migration de state_machine à state_machines-activerecord, qui est
l'évolution de la gemme

echarp committed:
Mise à jour vers redis 4.8, nouvelle méthode #exists? à la place de exists

echarp committed:
belongs_to est par défaut "not nullable" (depuis rails 5)

Ajout de "optional: true" pour les modèles qui permettent une valeur "null".

Document du comportement belongs_to:
https://sipsandbits.com/2015/09/21/whats-new-in-rails-5/#belongs_toisrequiredbydefault

echarp committed:
Complétion de clauses "group by" SQL pour fonctionner sur mysql standard

echarp committed:
image_tag a besoin d'un chemin plus complet

echarp committed:
Mécanisme de résolution standard pour templates rédaction/modération

Il y avait deux classes pour dynamiquement chercher des templates entre
les espaces de rédaction et modération, après tests et recherches il
semble que ça ne soit pas (ou plus?) nécessaire.

echarp committed:
Deux nouveaux templates pour gérer les téléchargements markdown depuis
les espaces de rédaction et modération

echarp committed:
Migration à rails 6.1

Tests et validations nécessaires, mais en pause car migration vers 7 prévue

echarp committed:
Migration initiale vers rails 7, avec gemmes locales et quelques pages non fonctionnelles

echarp committed:
La méthode #to_s des dates et heures renommée en #to_fs

echarp committed:
La gemme nio4r n'a plus besoin d'être bloquée à une anciennce version

echarp committed:
Mise à jour vers les sous-projets github

echarp committed:
Mise à jour à redis v5

echarp committed:
Dépréciation de méthode File.exists?

La méthode est maintenant nommée File.exist?

echarp committed:
Depuis rails 6 il est recommandé d'hériter d'une classe centrale gérée dans le projet

echarp committed:
Ajout de tests, surtout sur les contrôleurs de premier niveau,
qui sont à peu près touts couverts

echarp committed:
Déplacement de journal vers forum fonctionnel à nouveau

echarp committed:
Plus de tests d'intégration, pour 90% de couverture

echarp committed:
Chemin manquant pour accéder à des "partials" de news en modération

echarp committed:
La pluralisation nécessite un helper pour afficher le nombre

echarp committed:
L'API des errors a évolué, l'ancienne méthode "values" est dépréciée

echarp committed:
La mise à jour de la gemme doorkeeper nécessite une migration BdD

echarp committed:
MaJ script schéma BdD

echarp committed:
Plus de tests, pour une couverture à 95%

(en tous cas du code testé d'après la gemme simplecov...)

echarp committed:
Tous les contrôleurs admin sont testés, à 100%

echarp committed:
Rapide correction, pour pouvoir lister les suivis non assignés

(semble non utilisé dans l'IHM)

echarp committed:
Chemin manquant pour accéder à un "partials" de comment

Il pourrait y avoir d'autres cas, cela vient de la simplification de
recherche des chemins de partiaux, suite (ou grâce) à la migration à rails 7

echarp committed:
Plus de tests, les contrôleurs sont testés à plus de 99%

Il manque les contrôleurs d'API dans cette couverture...

echarp committed:
Test des threads de commentaires

echarp committed:
Un peu plus de tests de models

echarp committed:
La redirection vers duck duck go nécessite une permission explicite pour "sortir"

echarp committed:
Utilisation du serveur web par défaut des derniers rails: puma

Et mise à jour des dépendances

echarp committed:
Tests de contrôleurs à peu près completes, et propres

echarp committed:
Utilisation de Redis.new plutôt que de la variable globale $redis dans les tests

echarp committed:
Utilisation de Redis.new dans les tests de modèle aussi

Trim committed:
Add note to enable the DOCKER_BUILDKIT builder

echarp committed:
L'action "move" des liens n'existe pas
et ne devrait dont pas être référencée dans sa configuration

echarp committed:
Migration à rails 7.1

echarp committed:
Oubli configuration pour avoir des boutons sous forme de liens

echarp committed:
Test plus solide, et utilisation de double quote à certains endroits

echarp committed:
Utilisation de la gem normalize, plutôt qu'un module npm

echarp committed:
Suppression de node et npm

Les icônes feather sont maintenant gérés dans `vendor/assets/images`

echarp committed:
Suppression de références à node_modules

echarp committed:
Migration à rails 7.1.1

echarp committed:
Nettoyage Gemfile

Suppression de
* capistrano
* mo (obsolète, plutôt utiliser rubocop)
* pry-rails (la console irb normale est au moins aussi bien maintenant)
* sushi
* capybara
* selenium
* gctools (outil utilisé en production? Ne cible pas seulement ruby 2.1?)

Trim committed:
explain how to run linuxfr with rootless containers

nud committed:
Apply remarks from decaffeinate to the new javascript files

This commit applies the suggestions from decaffeinate to the newly
generated files so they are easier to read. The code still feels a
little weird though.

Suivi #3034

nud committed:
Fix scrolling to the bottom of the page when using shift+g

`$("body").attr("scrollHeight")` was returning undefined, so we get the
page height another way.

Trim committed:
use container engine agnostic file names as it works with Docker and Podman

Trim commented:
Pour la publication de la gem board-linuxfr sur Github Package Repository, j'ai réussi, mais c'est inutile: pour pouvoir télécharger la gem, il faut être connecté à Github, même si la gem est publique. Apparemment, [Github ne compte pas changer ça](https://github.com/orgs/community/discussions/26634#discussioncomment-8527086).

Comme on n'a pas prévu de passer à Gitlab (qui sait faire ça, en tout cas pour node), je vais demander à Bruno les accès aux gems sur rubygems.org.

echarp committed:
Bundle update

echarp committed:
Upgrade unicorn, carrierwave and acts_as_list, remove coffee gem

Trim committed:
update linuxfr-board container to use the board-sse-linuxfr.org gem version 1.0.0

Trim committed:
add to Container.md podman compose run, add IFS value for database entrypoint script

Trim commented:
J'ai publié aujourd'hui la nouvelle version de la gem `board-linuxfr` sous le nom de `board-sse-linuxfr.org` sur rubygems.org

J'avais essayé le Github Package Repository, mais on n'est obligé d'avoir un token Github pour télécharger la gem, même si elle est publique. C'était donc inutile :-1: 

J'ai mis à jour les fichiers container pour utiliser cette nouvelle gem et nouvelle version.

Enfin, j'ai rebasé la branche sur master pour récupérer les derniers changements fusionnés ces dernières semaines.

J'ai voulu contrôlé que les nouvelles fonctionnalités pour "les images externes bloquées" fonctionnent encore, mais j'ai l'impression qu'il y a un soucis avec le pipeline qui transforme le markdown en HTML (le tag `src` des balises `<img>` n'est pas mis à jour et les images n'apparaissent pas dans la liste des "images externes").

J'ai donc ajouté une tâche pour fixer ça avant de pousser sur alpha.

J'ai profité d'ajouter une deuxième tâche pour exécuter par défaut sans privilège root/réseau LinuxFr.

>
> * [ ] Corriger l'insertion des images pour utiliser le service img-linuxfr (la méthode wikify ne semble plus fonctionner, les images des liens restent sur la source)
> * [ ]  Utiliser par défaut les ports 8080 et 4443 pour ne plus avoir besoin de CAP réseau élevée

Trim committed:
URI.encode has been removed from Ruby 3

Image caching were not working because the Image model silently decided
all image were "internal": the URI.encode (aka URI.escape) method does
not exist anymore in Ruby 3, which raised an error which where ignored
by the "rescue" line.

Instead of doing "URI.parse(URI.encode(str))", we are just doing
"URI.join(str)", because the join method converts firt the str to
RFC3986.

As this method does not raise error, the "rescue" part has been removed
too.

Trim commented:
Ouf, ce n'était pas le pipeline de Markdown vers HTML qui était à modifier, mais juste le modèle Image qui utilisait la méthode `URI.encode` (alias `URI.escape`) qui n'existe plus en Ruby 3.

Trim committed:
linuxfr-img container can use bookworm to build the Go service

Trim committed:
Update nginx config and docker files to use Debian Bookworm with Ruby 3.1

Nginx proxy needed update on buffer size to avoid error about "upstream
sent too big header".

Trim committed:
Add nodejs back to linuxfr.org container which is used to uglify javascript assets

Trim commented:
Aujourd'hui, j'ai mis à jour le fichier `docker-compose.yaml` de la branche `master` pour inclure les commandes `healthchek` qui vérifient l'état des services.

J'ai aussi mis à jour sur la branche `master` les `Dockerfile` pour ne plus utiliser Debian Stretch, parce que j'y avais un problème qui m'empêchait d'exécuter correctement `npm ci` pour installer les dépendances (le processus ne rendait jamais la main). Tant qu'à faire, je les ai moderniser directement (les commandes RUN avec le bash "strict mode").

Du coup, pour la branche `rails7`, j'ai donc fait un rebase depuis `master` pour être à jour avec l'état de master (pour les déploiements avec container). Quelques commits qui modernisaient les Dockerfile/Containerfile ont disparu, parce qu'ils avaient déjà été backportés dans `master`.

J'ai ajouté aussi quelques commits sur la branche `rails7` pour:

1. faire en sorte que le service `linuxfr-img` utilise Debian Bookworm pour exécuter le service img-Linuxfr.org écrit en go
2. Réintégrer explicitement `nodejs` qui est nécessaire pour `uglify` les codes javascripts

Pour ce dernier point, LinuxFr n'a pas beaucoup de JavaScript et les moteurs modernes sont très efficaces pour avoir du code très réactif. Est-ce que vous pensez vraiment que c'est encore nécessaire ?

Je serai pour enlever la gem `uglifier` et donc `nodejs` et servir directement le JavaScript. Ça sera plus proche de la philosophie open source, car le code source sera lisible depuis la console JavaScript.

echarp commented:
Bonne idée. Moins de dépendance c'est mieux!

Trim committed:
compose: share gemfile from host

Trim committed:
replace uglifier gem by terser to minimize javascript files

The minimization of JavaScript files is useful, because LinuxFr.org
includes jQuery and jQeury UI and certainly don't use all the tools
provided by them.

The decaffeinate process to build JavaScript files from old
coffee script files has built EcmaScript 6 files.

As uglifier is not compatible with EcmaScript 6, it has been replaced by terser
which is recommended by the uglifier project itself.

Trim commented:
> Je serai pour enlever la gem `uglifier` et donc `nodejs` et servir directement le JavaScript. Ça sera plus proche de la philosophie open source, car le code source sera lisible depuis la console JavaScript.

Alors, j'ai essayé dans ma branche [remove-nodejs-and-uglifier](https://github.com/Trim/linuxfr.org/tree/remove-nodejs-and-uglifier) et en fait la taille a pas mal grossi, parce qu'on inclus jQuery v2.

Comme uglifier supprime les fonctions non utilisées de jQuery, il est quand même capable de bien réduire la taille totale de JavaScript à télécharger (200 Ko de moins).

Pour comparaison, voilà la taille du fichier JavaScript généré pour les 3 environnements:

| Description     | Branche rails7 | Branche remove-nodejs-and-uglifier | Production actuelle (ruby 2) |
| --------------- | -------------- | ---------------------------------- | ---------------------------- |
| Taille JS (kio) |        KO (**) |                             370 Ko |                       147 Ko |

(**): J'ai forcé l'utilisation de uglifier dans config/environments/develop.rb pour ce test, mais uglifier n'est pas compatible avec EcmaScript 6 et decaffeinate a justement fait des scripts au format EcmaScript 6

@oumph: si on enlève uglifier + nodejs de la stack, le fichier application.js grossis de 200Ko, est-ce que ce serait un problème pour la bande passante ?

echarp commented:
J'ai jeté un coup d'œil rapide, a priori il s'agit de jQuery avec jquery_ujs, c'est un peu indispensable en l'état :)

Zut zut, je pensais qu'on pourrait simplifier cette partie là.

Alors on peut envisager une autre direction: l'uglification se fait au moment de la précompilation des assets. On peut faire cette précompilation ailleurs que sur le serveur de prod. Et dans ce cas les assets précompilés peuvent être mis dans git ou poussés avec un autre mécanisme genre rsync... Je ne suis pas sûr que ça simplifie beaucoup.

Trim commented:
J'aurai voulu enlever jquery, parce qu'on utilise une version majeur obsolète (v2 au lieu de v3) et parce qu'avec JavaScript moderne on peut faire plein de choses sans jquery.

Mais ça sera un projet complet, parce que c'est jquery qui fait l'éditeur Markdown actuel de LinuxFr :)

Pour l'instant, l'uglifier ne fonctionne plus avec la branche rails7 etbdonc le seul choix est de livrer le JavaScript avec 200Ko de plus. 

Ce n'est pas non plus la mer à boire de nos jours ces 200Ko de plus 🙂 

i ce n'est pas ok pour l'hébergement, alors j'investirais plus de temps pour voir s'il y aurait une autre gem qui minifie et qui est compatible EcmaScript 6 (je crois en avoir vu un dans une des documentations que j'ai lu récemment).

Par contre, ce qui est bien, c'est que Nodejs ne devrait pas être un pré-requis pour l'environnement de dev, parce que uglifier y est désactivé pour pouvoir debugger.

Trim commented:
> Si ce n'est pas ok pour l'hébergement, alors j'investirais plus de temps pour voir s'il y aurait une autre gem qui minifie et qui est compatible EcmaScript 6 (je crois en avoir vu un dans une des documentations que j'ai lu récemment).

Ah je viens de tester [terser](https://github.com/ahorek/terser-ruby) pour remplacer la gem uglifier et ça fonctionne très bien et redonne une taille de 145Ko. Je met ce changement sur la branche `rails7` en attendant de voir si on enlève complètement nodejs et la minisation du code javascript.

echarp committed:
Bundle update (removes a small CVE)

echarp committed:
Passage à Haml 6+

La nouvelle version supprime quelques helper, dont `list_of` qui était
utilisée ici et là et qui est maintenant rajoutée dans
`application_helper`.

Méthodes supprimées:
  block_is_haml?, capture_haml, escape_once, find_and_preserve, flatten, haml_concat,
  haml_indent, haml_tag, haml_tag_if, html_attrs, html_escape, init_haml_helpers, is_haml?,
  list_of, non_haml, precede, succeed, surround, tab_down, tab_up, with_tabs

Référence: https://github.com/haml/haml/releases/tag/v6.0.0

echarp committed:
Passage to Sass 6+

Still maintained, will require an upgrade to dart-sass at some point.

echarp committed:
feat: Update gem dependencies and sitemap generator

This commit updates several gem dependencies in the Gemfile and
Gemfile.lock, including Rails and related gems. The version constraint
for the sitemap_generator gem has been removed, allowing the latest
version to be used.

The sitemap configuration has been updated to match the new usage of
sitemap_generator. The encoding comment and an extra line have been
removed, and the `add_links` method has been replaced with `create`.

Extra lines have been added before `sitemap.add` calls for readability.

Trim committed:
fix typos

Trim committed:
use custom HTTP public port to use rootless containers by default

This commit introduces two new global constants MY_PUBLIC_URL and
IMG_PUBLIC_URL which should be used now to build URLs.

For alpha and production Ruby environments, they are fixed with their real
values directly in the `config/environments/*.rb` files.

For the development Ruby environment, they are defined by default to
`http://dlfp.lo` and `http://image.dlfp.lo`. You can customize either by
updating the `config/environments/development.rb` file or by setting the
*process* environment variables `DOMAIN`, `DOMAIN_HTTP_PORT`, `IMAGE_DOMAIN`
and `IMAGE_DOMAIN_HTTP_PORT`.

For containers, you can set these variables in the `deployment/default.env`
file.

Trim commented:
Quand j'ai configuré une nouvelle machine, j'ai ce message d'avertissement à la création des bases de données:

> $ podman-compose exec linuxfr.org bin/rails db:setup
> podman-compose version: 1.0.6
> ['podman', '--version', '']
> using podman version: 4.3.1
> podman exec --interactive --tty linuxfrorg-rails7_linuxfr.org_1 bin/rails db:setup
> Created database 'linuxfr_rails'
> Created database 'linuxfr_test'
> **NOTE: nokogumbo: Using Nokogiri::HTML5 provided by Nokogiri. See https://github.com/sparklemotion/nokogiri/issues/2205 for more information.**
> exit code: 0

echarp commented:
Le warning nokogiri semble être lié à des dépendances intermédiaires. À voir si ça évolue avec les mises à jour.

Pour les credentials, c'est une autre évolution de rails, qui sont passés sur un autre mécanisme de stockage des secrets. Comme on utilise, il me semble, des variables d'environnement passées par le serveur web, on devrait pouvoir remplir sans utiliser... ou quelque chose de plus approprié, à voir ce qu'on veut faire.

Trim commented:
Le 22 septembre 2024 23:14:33 GMT+02:00, echarp ***@***.***> a écrit :
>Le warning nokogiri semble être lié à des dépendances intermédiaires. À voir si ça évolue avec les mises à jour.

Oui, j'ai vu, ça semble venir de sanitizer 5 qui est installé par notre outil html-pipeline-linuxfr. On peut mettre à jour notre outil pour passer à sanitizer 6 qui a mis à jour nokogiri.

Comme sanitizer fait un job assez critique, j'aimerai bien le mettre à jour en même temps que rails.


>Pour les credentials, c'est une autre évolution de rails, qui sont passés sur un autre mécanisme de stockage des secrets. Comme on utilise, il me semble, des variables d'environnement passées par le serveur web, on devrait pouvoir remplir sans utiliser... ou quelque chose de plus approprié, à voir ce qu'on veut faire.
>

J'ai essayé de regarder hier et je crois qu'on l'utilise pour générer des secrets aléatoires au démarrage du service. Après, ça ne bloque pas cette mise à jour, ça sera à penser pour la mise à jour vers rails 7.2.

Trim commented:
>> Pour les credentials, c'est une autre évolution de rails, qui sont passés sur un autre mécanisme de stockage des secrets. Comme on utilise, il me semble, des variables d'environnement passées par le serveur web, on devrait pouvoir remplir sans utiliser... ou quelque chose de plus approprié, à voir ce qu'on veut faire.
>
> J'ai essayé de regarder hier et je crois qu'on l'utilise pour générer des secrets aléatoires au démarrage du service. Après, ça ne bloque pas cette mise à jour, ça sera à penser pour la mise à jour vers rails 7.2.

Alors je me suis trompé, les valeurs sont fixes pour les environnements development et tests. Comme tu l'as relevé, ce sont des variables d'environnement pour la production.

J'ai lu un peu la documentation du [nouveau système](https://guides.rubyonrails.org/security.html#custom-credentials) et c'est intéressant parce que:

1. ça évite de mettre des clés privées / secrets dans les environnements de variables (ces variables sont accessibles depuis /proc).
    * bon, LinuxFr est hébergé sur ses propres serveurs, c'est du coup moins important. Dans ce cas, ça permet d'éviter qu'un bug dans un autre logiciel qui tourne sur le même serveur révèle les secrets des variables d'environnement, par exemple dans un dump.
2.  on peut versionner la configuration complète dans le git: les secrets sont chiffrés avec ce système et seul les personnes / logiciels qui ont accès à la clé privée "master.key" peut révéler et éditer les secrets.

Cette évolution est à réfléchir, je ne sais pas si elle est souhaitée par les admins de LinuxFr. On pourra le faire dans un second temps en se coordonnant avec eux.

echarp committed:
chore: upgrade to Rails 7.2 and improve secrets handling

- Upgraded Rails from 7.1 to 7.2 in Gemfile and related files.
- Updated config/initializers/secrets.rb to load secrets from environment
  variables if present, falling back to credentials if not.
- Ensured that `secret_key_base`, `push_secret`, and `devise_pepper` are
  loaded securely.
- Updated config/secrets.yml.sample to reflect the changes.

echarp committed:
chore: upgrade to Rails 8.0

echarp committed:
chore: upgrade to Rails 8.1

echarp committed:
chore: upgrade to Rails 8.1.2

echarp committed:
doc: update installation instructions for Rails 8 and Debian Trixie

- Target Debian Trixie (stable)
- Update MySQL/MariaDB package names
- Simplify Bundler installation (now included with Ruby)
- Improve document structure and add syntax highlighting

Trim committed:
nginx container: fix empty alias error

Trim commented:
Je viens de `rebase` depuis `master` pour aider à fusionner la migration à Rails8 #399, mais je vois que ça n'a pas vraiment aidé :-/

Trim committed:
update containers with Debian Trixie as suggested in INSTALL.md

Trim commented:
@echarp merci pour la mise à jour vers Rails 8 :+1: 

J'ai juste une question par rapport au `Gemfile`: il y a 2 gems qui pointent vers tes forks github (`french_rails` et `html-pipeline-linuxfr`). Comment penses-tu que l'on devrait gérer la migration ?

La faire en 2 étapes ? D'abord garder comme ça le Gemfile, puis intégrer les changements dans les repo de l'organisation `linuxfrorg` ?

echarp commented:
Non non, j'ai fait 2 changements sur ces gemmes, j'attends juste qu'elles soient validées pour pointer vers les mêmes que d'habitude

Trim committed:
add notes about CSS compilation

echarp committed:
chore(deps): update dependencies to related linuxfr gems

For french_rails and html-pipeline-linuxfr

echarp committed:
Merge branch 'master' of github.com:linuxfrorg/linuxfr.org into rails7

Trim commented:
Je viens de voir que la gem `sassc-rails` est déjà [abandonnée](https://github.com/sass/sassc-rails) et qu'il faudrait passer sur [dart-sass](https://sass-lang.com/dart-sass/).

echarp commented:
> Je viens de voir que la gem `sassc-rails` est déjà [abandonnée](https://github.com/sass/sassc-rails) et qu'il faudrait passer sur [dart-sass](https://sass-lang.com/dart-sass/).

Pas la peine de se presser, il y a plein d'éléments qu'il faudra mettre à jour, notamment passage à importmap et propshaft.

Mais d'abord n'avoir qu'une branche de travail.

On pourra aussi passer un outil pour standardiser tout le code (standardrb fait ça très bien).

echarp committed:
chore(deps): updated to latest rails 8.1.2.1

Trim committed:
Merge branch 'main' into rails7

Trim commented on app/assets/javascripts/chat.js:
```diff
@@ -0,0 +1,344 @@
+  right_highlitizer(event) {
+    const time = $(event.target).data("clockTime");
+    const index = $(event.target).data("clockIndex");
+    if ((time.length = 5)) {
```
ah tiens, c'est bien vu comme problème, je vais regarder s'il existe aussi sur main dans le coffeescript correspondant.

Trim committed:
Merge branch 'main' into rails7

Conflicts:
        app/assets/javascripts/chat.coffee

Resolve conflicts by applying fix of commit
52ac1191c5637b97b0c32edb0fcf74c5f7b50810 which has been made on
coffeescript to the corresponding JavaScript code.

Trim committed:
application.js: remove unreachable return statement
